synapse/http/proxyagent.py

```diff
@@ -213,5 +228,11 @@ def parse_host_port(hostport, default_port=None):
             # the thing after the : wasn't a valid port; presumably this is an
             # IPv6 address.
             pass
-
     return hostport, default_port
+
+
+def parse_scheme_host(schemehost, default_scheme=None):
```

Comment:
Please add type info and a docstring to any new methods!

I think this is pretty much `urllib.parse.urlparse(urlstring, scheme=default_scheme)[0:1]` so I'm not sure this method is necessary?

Reply:
I actually used urlparse in the first instance of this PR but that has going horribly wrong when being given a think of the form `proxy.foo.bar:9000`. Which then got parsed as `scheme=proxy.foo.bar`, `url:9000` *even though explicitly passing in the default port*. Ugh. You can still see my frustrated commit here, changing it to the current form. :D